otwcode/otwarchive
# AO3-6464 Allow ruby, rt, and rp HTML tags where HTML is allowed

# Pull Request Checklist

  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6464

## Purpose

The `ruby`, `rt`, and `rp` tags are used to represent [ruby annotation](https://en.wikipedia.org/wiki/Ruby_character). With these tags, the small text marked with `rt` will be located above the base text marked with `ruby`. Although there are various complexities that come with more advanced features of ruby annotation (e.g. the tabular syntax, CSS rules to control how to render ruby characters, etc), the intention behind the linked Jira issue seems to be to support the most simple (and common) use cases.

## Testing Instructions

To start with, please try posting a new work with the new tags (and with the usual required fields).

Suggested content: `<ruby>BigText<rp> (</rp><rt>short_text</rt><rp>)</rp></ruby>`

This should work in other places where the limited HTML is allowed (like comments).

## References

`rb` is not included here (thus removed by the sanitizer when entered). I think this could potentially be revisited if and when https://github.com/whatwg/html/issues/1771 is resolved. See also the linked Jira issue for related discussion.

## Credit

Potpotkettle (they/them)
